internal/controller/pkg/resolver/reconciler.go

```diff
@@ -298,14 +274,64 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// no packagePullSecrets are set. Settings can be modified manually
 	// after dependency creation to address this.
 	pack.SetName(xpkg.ToDNSLabel(ref.Context().RepositoryStr()))
-	pack.SetSource(fmt.Sprintf(packageTagFmt, ref.String(), addVer))
+
+	format := packageTagFmt
+	if strings.Contains(addVer, "sha256") {
```

Comment:
```suggestion
	if strings.HasPrefix(addVer, "sha256:") {
```

Let's be a bit more strict. We wouldn't want a version like `v0.0.0-sha256-<digest>` to be processed as a digest :)